### PR TITLE
Add AppStream metadata and `.desktop` file for Linux desktop integration

### PR DESCRIPTION
**Pull request by Calinou**
Tested locally on my Fedora 33 system, it works:

KRunner showing ImHex

- Rename the binary target from `ImHex` to `imhex` for easier invocation on case-sensitive filesystems.
- Add a 256×256 PNG icon to be installed on Linux (upscaled using waifu2x).

This supersedes/closes #40 by also providing CMake install functionality.

Adding Linux desktop integration files eases packagers' job and also makes it possible to package ImHex for [Flathub](https://flathub.org).

### CODE REVIEW

**Review by umarcor — APPROVED** [review]
IMHO, this could be rebased on #40 and renaming the binary should be done in a different PR (or commit, at least). Anyway, if maintainers are ok, it's all good.

@Calinou, please add `Close #40` to your comment :wink:

File: dist/linux/net.werwolv.ImHex.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  <releases>
+    <release version="1.5.0" date="2020-12-06"/>
+    <release version="1.4.0" date="2020-11-28"/>
+    <release version="1.3.1" date="2020-11-23"/>
+    <release version="1.3.0" date="2020-11-22"/>
+    <release version="1.2.0" date="2020-11-21"/>
+    <release version="1.1.0" date="2020-11-17"/>
+    <release version="1.0.0" date="2020-11-12"/>
+  </releases>
+  <update_contact>hugo.locurcio@hugo.pro</update_contact>
```
Comment (Thog) [inline dist/linux/net.werwolv.ImHex.appdata.xml:65]: Should maybe be @WerWolv's email here?

**Review by Thog — APPROVED** [review]
lgtm but need a rebase

File: CMakeLists.txt [hunk 2]
```diff
@@ -91,14 +91,17 @@ add_executable(ImHex
...
 endif (WIN32)
 
 if (UNIX)
-    target_link_libraries(ImHex libglfw.so libmagic.so libcrypto.so libdl.so libcapstone.so ${llvm_demangle_lib} ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
+    target_link_libraries(imhex libglfw.so libmagic.so libcrypto.so libdl.so libcapstone.so ${llvm_demangle_lib} ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
 endif (UNIX)
 
-install(TARGETS ImHex DESTINATION bin)
+install(TARGETS imhex DESTINATION bin)
+install(FILES icon.png DESTINATION share/icons/hicolor/256x256/apps RENAME imhex.png)
```
Comment (filcab) [inline CMakeLists.txt:105]: You probably only want to do this on certain OS.

Comment (Calinou) [reply CMakeLists.txt:105]: How do other projects typically handle this? Could you link to some examples?

In most CMake projects I've seen, it's expected that the install step is only used on Linux/*BSD, not Windows or macOS.

Comment (umarcor) [reply CMakeLists.txt:105]: It's used on MSYS2 too. See https://github.com/WerWolv/ImHex/blob/master/msys2/PKGBUILD#L42. That's Windows but with a Unix path syntax, tho. Therefore it's ok.

Comment (filcab) [reply CMakeLists.txt:105]: I guess that makes some sense, though wouldn't programs usually do an install step for copying stuff to a macOS `*.app` or to setup a Windows install dir with everything?
I guess it's possible to do that with a different script (my experience is with unix-style installations like llvm (even on windows), so I don't know the usual stuff applications do.

Comment (umarcor) [reply CMakeLists.txt:105]: IMHO, it's good enough as Hugo proposed it. There might be some issue in the future with some specific packaging for macOS or Windows. Yet, let's let the packagers that find those issues in the future to speak up and raise the concern. Luckily, none of this changes is immutable.

**Comment by umarcor** [comment]
@Calinou, while rebasing, remember to update https://github.com/WerWolv/ImHex/blob/master/.github/workflows/build.yml#L174

**Comment by Calinou** [comment]
> @Calinou, while rebasing, remember to update master/.github/workflows/build.yml#L174

Done :slightly_smiling_face:

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -171,4 +171,4 @@ jobs:
 
     - name: ▶️ Test
```
Comment (toot-gerly) [inline .github/workflows/build.yml:172]: ع

Comment (toot-gerly) [reply .github/workflows/build.yml:172]: Tootgerly

**Review by toot-gerly — APPROVED** [review]

**Comment by Mailaender** [comment]
Icon looks blurry. You may get better result if you disable any filters while doing so.

**Comment by Calinou** [comment]
@Mailaender I already use an AI-based method to upscale it, and it's a temporary icon anyway. I won't spend too much effort into it considering the icon will be replaced soon :slightly_smiling_face:

**Comment by Mailaender** [comment]
Well, my point is that AI is overdone here. No interpolation gives best results with basic pixelart:

image

**Comment by Calinou** [comment]
@Mailaender I added the icon you provided to my PR.

**Comment by Mailaender** [comment]
Awesome. Looks like you need to [rebase your pull request](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request).

**Comment by Calinou** [comment]
> Awesome. Looks like you need to [rebase your pull request](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request).

It looks like the CMake setup has changed entirely since I opened this pull request. I can't test this locally since the [build step fails on my machine](https://github.com/WerWolv/ImHex/issues/169).

**Comment by Mailaender** [comment]
Try to rebase this on v0.7.0 and see my @openSUSE .spec file: https://build.opensuse.org/request/show/874069

**Comment by hasan517** [comment]
100009190788216

**Review by hasan517 — APPROVED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -93,14 +93,17 @@ add_executable(ImHex
         resource.rc
         )
 
-target_link_directories(ImHex PRIVATE ${LLVM_LIBRARY_DIR})
+target_link_directories(imhex PRIVATE ${LLVM_LIBRARY_DIR})
```
Comment (Mailaender) [inline CMakeLists.txt:96]: This was already renamed to lower case in https://github.com/WerWolv/ImHex/commit/4b59d57f899823f595087ff7e0289997d72c5560

**Review by hasan517 — COMMENTED** [review]
Abdi

**Comment by WerWolv** [comment]
Thanks a lot, this has been added for the .deb and AppImage package now

**Comment by Mailaender** [comment]
It looks like you didn't merge this. Where can I find `net.werwolv.ImHex.appdata.xml`?

**Comment by WerWolv** [comment]
@Mailaender I didn't add that since it wasn't used anywhere anyways, just the .desktop file was. 
What would it be useful for?

**Comment by Mailaender** [comment]
The idea is to have this upstream, because it is used by distributions universally not just @flathub. Also it is your applications. You should decide how it is presented in AppStream store applications. See https://github.com/flathub/flathub/pull/3195/files for my current version.

**Pull request closed without merging**